websurfx/config.lua

```diff
@@ -5,7 +5,7 @@ threads = 10 -- the amount of threads that the app will use to run (the value sh
 
 -- ### Server ###
 port = "8080" -- port on which server should be launched
-binding_ip = "127.0.0.1" --ip address on the which server should be launched.
+binding_ip = "0.0.0.0" --ip address on the which server should be launched.
```

Comment:
Also, I would like to suggest not changing this file because this project is packaged in other packaging formats as well. When users will install these from these formats, they can mistakenly expose their IP address on the Internet because they might not know about it. Or they thought the default config was fine for running the search engine, which is, as you know, bad for privacy. So I would recommend not updating the file but providing a sample config with details on what to update and how to run the project for development in a separate page on the docs.

Reply:
Wdym by exposing IP address ?
Because it doesn't it just tell to the system to bind on all available IPV4 address on the local machine.
And there isn't other workarround to make it work on docker (production or development) also in my opinion theses kind of config should be ENV variables. Or passed as argument to the program directly.

Reply:
> Wdym by exposing IP address ? Because it doesn't it just tell to the system to bind on all available IPV4 address on the local machine. And there isn't other workarround to make it work on docker (production or development) also in my opinion theses kind of config should be ENV variables. Or passed as argument to the program directly.

Yes, true, but then it also tells to bind on the IP address that was assigned to the system by the network. Let's say if it was a Public IP, then this could be an issue as you know this will accidentally expose the search engine publicly. 

Also, I agree with you that env variables are a good idea, but they tend to be not very great when it comes to maintenance, flexibility, and customizability. For example, if you were going to host the search engine on four devices then you will need to put four different variables on each machine, which is, as you know, very painful. Moreover, if we were to let's say tomorrow propose a breaking change, then the user will have to set the new env variables on all four machines separately. So by using a config file and also one written in a programming language, it provides a good amount of flexibility. Since our config file is written in Lua, you can take advantage of loops, conditionals, etc. to write only one config which covers all the devices. Now if a change were to occur then this will require only the file needs to be changed. Also, there are benefits as well because of it being written in a programming language. I hope you got my point :slightly_smiling_face: .

Reply:
If you install it on a dedicated server, isn't the goal to have it publicly available?
You wont be able to access to it if you only bind to 127.0.0.1.
As for the ENV variable, it's industry standard to be able to set the HOST and the PORT.
And that most likely never causes a breaking change.
But whatever.